src/party_menu.c

```diff
@@ -0,0 +1,7231 @@
+        }
+    }
+}
+
+bool16 sub_81B2134(struct Pokemon *mon)
+{
+    if (GetMonData(mon, MON_DATA_IS_EGG) != TRUE && sub_802C908(GetMonData(mon, MON_DATA_SPECIES)) != FALSE)
+        return TRUE;
+    return FALSE;
+}
+
+// Dodrio Berry Picking select?
+
+bool16 sub_81B2164(struct Pokemon *mon)
+{
+    if (GetMonData(mon, MON_DATA_IS_EGG) != TRUE && GetMonData(mon, MON_DATA_SPECIES) == SPECIES_DODRIO)
+        return TRUE;
+    return FALSE;
+}
+
+bool8 sub_81B218C(u8 slot)
+{
+    if (!((gUnknown_0203CEC8.unkE >> slot) & 1))
+        return FALSE;
+    return TRUE;
+}
+
+void sub_81B21AC(u8 taskId, u8 slot)
+{
+    if (sub_81B218C(slot) == TRUE)
+    {
+        PlaySE(SE_SELECT);
+        gSpecialVar_0x8004 = slot;
+        sub_81B12C0(taskId);
+    }
+    else
+    {
+        PlaySE(SE_HAZURE);
+        sub_81B1B5C(gText_PkmnCantParticipate, 0);
+        schedule_bg_copy_tilemap_to_vram(2);
+        gTasks[taskId].func = sub_81B1C1C;
+    }
+}
+
+void sub_81B2210(u8 taskId)
+{
+    sub_81B1B5C(gText_CancelParticipation, 1);
+    schedule_bg_copy_tilemap_to_vram(2);
+    gTasks[taskId].func = sub_81B2248;
+}
+
+void sub_81B2248(u8 taskId)
+{
+    if (sub_81B1BD4() != TRUE)
+    {
+        sub_81B334C();
+        gTasks[taskId].func = sub_81B227C;
+    }
+}
+
+void sub_81B227C(u8 taskId)
+{
+    switch (Menu_ProcessInputNoWrapClearOnChoose())
+    {
+        case 0:
+            gSpecialVar_0x8004 = 7;
+            sub_81B12C0(taskId);
+            break;
+        case MENU_B_PRESSED:
+            PlaySE(SE_SELECT);
+        case 1:
+            gTasks[taskId].func = sub_81B1C1C;
+            break;
+    }
+}
+
+u8 CanPartyPokemonLearnTMTutor(struct Pokemon *mon, u16 item, u8 tutor)
+{
+    u16 move;
+    
+    if (GetMonData(mon, MON_DATA_IS_EGG) != FALSE)
+        return CANNOT_LEARN_MOVE_IS_EGG;
+    if (item >= ITEM_TM01_FOCUS_PUNCH)
+    {
+        if (CanMonLearnTMHM(mon, item - 33) != FALSE)
+            move = ItemIdToBattleMoveId(item);
+        else
+            return CANNOT_LEARN_MOVE;
+        do {} while (0); // :morphon:
+    }
+    else if (sub_81B2370(GetMonData(mon, MON_DATA_SPECIES), tutor) == FALSE)
+    {
+        return CANNOT_LEARN_MOVE;
+    }
+    else
+    {
+        move = sub_81B2360(tutor);
+    }
+    
+    if (pokemon_has_move(mon, move) == TRUE)
+        return ALREADY_KNOWS_MOVE;
+    return CAN_LEARN_MOVE;
+}
+
+u16 sub_81B2360(u8 tutor)
+{
+    return gUnknown_0861500C[tutor];
+}
+
+bool8 sub_81B2370(u16 species, u8 tutor)
+{
+    if (gUnknown_08615048[species] & (1 << tutor))
+        return TRUE;
+    return FALSE;
+}
+
+void sub_81B239C(u8 a)
+{
+    u8 i;
+    
+    switch (a)
+    {
+        case 0:
+            InitWindows(gUnknown_08615810);
+            break;
+        case 1:
+            InitWindows(gUnknown_08615850);
+            break;
+        case 2:
+            InitWindows(gUnknown_08615890);
+            break;
+        default:
+            In
```

Comment:
5492 & 5514: Use -1 and -2 (should use constants but they don't exist yet)

Reply:
when I tried it didn't match, maybe some other way of using them may work but simply substituting in -1 and -2 will not